apis/python/src/tiledbsoma/_tdb_handles.py

```diff
@@ -166,6 +166,11 @@ def _opener(
         """Opens and returns a TileDB object specific to this type."""
         raise NotImplementedError()
 
+    def reopen(self, mode: options.OpenMode) -> clib.SOMAObject:
+        return self._handle.reopen(
+            clib.OpenMode.read if mode == "r" else clib.OpenMode.write
```

Comment:
if someone calls `reopen(mode="foobar")` it will treat it as synonymous with `reopen(mode="w")`.  Seems like we need better error checking.